spikeinterface/preprocessing/silence_periods.py

```diff
@@ -0,0 +1,132 @@
+    """
+    Silence user-defined periods from recording extractor traces. By default, 
+    periods are zeroed-out (mode = 'zeros'). This is only recommended 
+    for traces that are centered around zero (e.g. through a prior highpass
+    filter); if this is not the case, you can also fill the periods with noise.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor to remove artifacts from
+    list_periods: list of lists/arrays
+        One list per segment of tuples (start_frame, end_frame) to silence
+
+    mode: str
+        Determines what periods are replaced by. Can be one of the following:
+            
+        - 'zeros' (default): Artifacts are replaced by zeros.
+
+        - 'noise': The periods are filled with a gaussion noise that has the
+                   same variance that the one in the recordings, on a per channel
+                   basis
+    num_chunks_per_segment: int
+        Number of chunks per segment for random chunk, by default 20
+    chunk_size : int
+        Size of a chunk in number for random chunk, by default 10000
+    seed : int
+        Random seed for random chunk, by default None
+
+    Returns
+    -------
+    silence_recording: SilencedPeriodsRecording
+        The recording extractor after silencing some periods    
+    """
+    name = 'silence_periods'
+
+    def __init__(self, recording, list_periods, mode='zeros', num_chunks_per_segment=20,
+                 chunk_size=10000, seed=None):
+
+        available_modes = ('zeros', 'noise')
+        num_seg = recording.get_num_segments()
+
+
+        if num_seg == 1:
+            if isinstance(list_periods, (list, np.ndarray)) and not np.isscalar(list_periods[0]):
+                # when unique segment accept list instead of of list of list/arrays
+                list_periods = [list_periods]
+
+        # some checks
+        assert isinstance(list_periods, list), "'list_periods' must be a list (one per segment)"
+        assert len(list_periods) == num_seg, "'list_periods' must have the same length as the number of segments"
+        assert all(isinstance(list_periods[i], (list, np.ndarray)) for i in range(num_seg)), \
+            "Each element of 'list_periods' must be array-like"
+
+        for periods in list_periods:
+            if len(periods) > 0:
+                assert np.all(np.diff(np.array(periods), axis=1) > 0), "t_stops should be larger than t_starts"
+                assert all(e < s for (_, e), (s, _) in zip(periods, periods[1:])), "Intervals should not overlap"
+
+        assert mode in available_modes, f"mode {mode} is not an available mode: {available_modes}"
+
+        if mode in ['noise']:
+            noise_levels = get_noise_levels(recording, return_scaled=False, 
+                                    num_chunks_per_segment=num_chunks_per_segment,
+                                    chunk_size=chunk_size, concatenated=True, seed=seed)
```

Comment:
For reproducability we should be able to exterbnally give the noise.

Reply:
Externally give the noise? The seed is not enough?

Reply:
I unified with other functions by passin `**random_chunk_kwargs`